eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -22,15 +22,15 @@ jobs:
           VCPKG_DEFAULT_TRIPLET: 'x64-windows-static'
           CMAKE_GENERATOR: 'Visual Studio 16 2019'
           CMAKE_GENERATOR_PLATFORM: x64
-          build.args: ' -DTRANSPORT_PAHO=ON'
+          build.args: ' -DTRANSPORT_PAHO=ON -DTRANSPORT_CURL=ON'
           test_type: 'iot'
           os: 'win'
         MacOS_x64_with_iot_samples:
           Pool: Azure Pipelines
           OSVmImage: 'macOS-10.15'
           vcpkg.deps: 'cmocka paho-mqtt'
           VCPKG_DEFAULT_TRIPLET: 'x64-osx'
-          build.args: ' -DTRANSPORT_PAHO=ON'
+          build.args: ' -DTRANSPORT_PAHO=ON -DTRANSPORT_CURL=ON'
           test_type: 'iot'
           os: 'macos'
         # Disable live testing for these matrix entries until we have something to
```

Comment:
We may want to try to use other legs for the "HTTP-based" samples, rather than change the IoT ones.

Reply:
@danieljurek what should our strategy be here?